azurerm/internal/services/loganalytics/validate/log_analytics_linked_service_automation_id.go

```diff
@@ -0,0 +1,17 @@
+package validate
+
+import "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+
+// IsAutomationAccountID parses a resource ID and returns a bool indicating if it is a valid AutomationAccountID
+func IsAutomationAccountID(input string) bool {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return false
+	}
+
+	if _, err = id.PopSegment("automationAccounts"); err != nil {
+		return false
+	}
+
+	return true
```

Comment:
@WodansSon rather than including an ID from outside of it's package, can we generate the "AutomationAccountId" in the automation package and reference that's validation function here, rather than adding this in? Else we'll end up with duplicates all over the place

Reply:
I'll open another PR to fix that bit...  thanks...